zebra-network/src/peer_set/set.rs

```diff
@@ -497,6 +508,15 @@ where
                     // - check for any errors that happened right after the handshake
                     trace!(?key, "got Change::Insert from Discover");
                     self.remove(&key);
```

Comment:
This `remove()` call is now incorrect.

For security/reliability reasons, I think old connections with exactly the same IP and port should stay, rather than replacing them with new connections. This matches the behaviour of the IP-based security fix in this PR.

Otherwise, two nodes could get in a loop where they initiate new connections, and they continually replace the other node's connection.

```suggestion
```